File: acode/abc255/c/main.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
  ll x, a, d, n;
  cin >> x >> a >> d >> n;
  if (d == 0) {
    cout << abs(a-x) << endl;
    return 0;
  }

  // no need
  //if (d < 0) {
  //  a = a+d*(n-1);
  //  d = -d;
  //}

  ll i = (x-a)/d;
  auto f = [&](ll i) {
    if (i < 0) i = 0;
    if (i >= n) i = n-1;
    return a+d*i;
  };
  ll ans = abs(f(i)-x);
  ans = min(ans, abs(f(i+1)-x));
  cout << ans << endl;
  return 0;
}

File: acode/abc255/c/main.py
X, A, D, N = list(map(int, input().split()))

# you need this because you are comparing with the value to take care of minus version and plus version. However, if you consider index like the video tutorial, you don't need to compare like this
mi = min(A, A + (N-1)*D)
ma = max(A, A + (N-1)*D)

if X <= mi:
    print(mi-X)
    exit()
if ma <= X:
    print(X-ma)
    exit()


base = X - A

diff = base % D

if diff == 0:
    print(0)
else:
    print(min(abs(diff-D), abs(diff)))



